raft.go

```diff
@@ -369,12 +369,12 @@ func (rn *Node) Destroy() error {
 	}
 	rn.logger.Debug("Successfully removed self from canoe cluster")
 
-	if rn.running {
+	if rn.running.IsSet() {
```

Comment:
Hrm. Yeah you're right. I'll look back over and see about using mutexes possibly - or some other construct. Or a check and set possibly